code/src/pages/products/ProductsPage.constants.tsx

```diff
@@ -0,0 +1,19 @@
+import { FormattedMessage } from "react-intl";
+
+export const sortOptions = [
+  {
+    value: "recommended",
+    label: <FormattedMessage id="productsDefault.label" />

```

Comment:
Store only id here and render jsx inside `.tsx` file

Reply:
> Store only id here and render jsx inside `.tsx` file

This approach for type `ReactNode` also make sense. not see the reason to change it.
Is It improve something ?

Reply:
Definitely yes. If we store only translation key, we will avoid duplicating `FormattedMessage` component multiple times and use it only once. Then if we want to change props for `FormattedMessage`, it would be easier to refactor in the future

Reply:
> Definitely yes. If we store only translation key, we will avoid duplicating `FormattedMessage` component multiple times and use it only once. Then if we want to change props for `FormattedMessage`, it would be easier to refactor in the future

I'm not understand. What do you suggest to do in this case ?

Reply:
Store only translation key in constants and rename it to `.ts` extension

Reply:
> Store only translation key in constants and rename it to `.ts` extension

I already define `ReactNode` type for it. I won't to rewrite types everywhere with this approach. Because for now it look nice and don't see a reason to pass separate key value in constants

Reply:
I don't agree with you. Let's rewrite it. If it is hard for you, i can suggest my help

Reply:
> I don't agree with you. Let's rewrite it. If it is hard for you, i can suggest my help

We can leave it. and ask out expert. It just your opinion don't force it